docs/tokenomics/nns/proposal-requirements.md

```diff
@@ -0,0 +1,206 @@
+
+* **DissolveState**: at any time, at most one of WhenDissolved and DissolveDelay are specified.
+** WhenDissolved (timestamp)
+When the dissolve timer is running, this stores the timestamp in seconds from the Unix epoch, at which point the neuron becomes dissolved. At any time while the neuron is dissolving, the neuron owner may pause dissolving, in which case DissolveDelay will get assigned to WhenDissolved minus the timestamp when the action is taken.
+** DissolveDelay (duration)
+When the dissolve timer is stopped, this stores how much time the dissolve timer will be started with. It can be eight years at most. At any time while in this state, the neuron owner may (re)start dissolving, in which case WhenDissolved will get assigned to the timestamp when the action is taken plus DissolveDelay.
+
+* **Maturity (positive number – percent)**: the maturity of a neuron, which determines its ability to spawn a new neuron and corresponding locked balance of newly minted ICP in expectation equal to this value in percent of the stake of the spawning neuron. When new neurons are created, their maturity is zero. When neurons vote, over time the NNS increases their maturity to reward them.
+
+* **Follow relationships (mapping from topic to list of followers, private)**: a neuron can be configured to vote automatically by following other neurons on a topic-by-topic basis. For any valid topic, a list of followers can be specified, and the neuron will follow the vote of a majority of the followers on a proposal with a type belonging to that topic. If a null topic is specified, this acts as a catch-all that enables the neuron to follow the vote of followees where a rule has not been specified.
+
+* **Recent votes (public)**: a record of recent votes is maintained. This can provide a guide for those wishing to evaluate whether to follow a neuron or how their followers are voting.
+
+* **NotForProfit (boolean)**: whether this neuron is “not for profit,” making it dissolvable by voting.
+
+The following attributes can be computed:
+
+* **Age (seconds)** (computed from AgingSince and current time): the period that has elapsed since the neuron was created or last stopped dissolving. Conceptually, whenever a neuron starts dissolving, then its age is reset to zero and remains zero while it is dissolving. If a dissolving neuron has been turned off, the current time becomes the effective neuron creation date to calculate the age.
+* **State (LOCKED or DISSOLVING or DISSOLVED)** (computed from DissolveState and the current time):
+
+    - LOCKED: In this state, the neuron is locked with a specific DissolveDelay. It accrues age by the passage of time and it can vote if DissolveDelay is at least six months. The method `start_dissolving` can be called to transfer the neuron to the DISSOLVING state. The method `increase_dissolve_delay` can be used to increase the dissolve delay without affecting the state or the age of the neuron.
```

Comment:
```suggestion
    - 
    NOT_DISSOLVING: In this state, the neuron is locked with a specific DissolveDelay. It accrues age by the passage of time and it can vote if DissolveDelay is at least six months. The method `start_dissolving` can be called to transfer the neuron to the DISSOLVING state. The method `increase_dissolve_delay` can be used to increase the dissolve delay without affecting the state or the age of the neuron.
```
This should be applied everywhere: this is now called NOT_DISSOLVING in the code and "in English" we call this non-dissolving sometimes, but let's try to not use "locked" as this might be confusing